### ISSUE TITLE
Docassemble latest docker image error.

### ISSUE THREAD

**Issue by japrogramer** [jhpyle/docassemble#646]
```
Traceback (most recent call last):
  File "/usr/share/docassemble/local3.10/lib/python3.10/site-packages/docassemble/webapp/update.py", line 269, in check_for_updates
    inst_vers = version.parse(installs[package.id].packageversion)
KeyError: 5

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/usr/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/usr/share/docassemble/local3.10/lib/python3.10/site-packages/docassemble/webapp/update.py", line 736, in <module>
    main()
  File "/usr/share/docassemble/local3.10/lib/python3.10/site-packages/docassemble/webapp/update.py", line 709, in main
    check_for_updates(start_time=start_time, invalidate_cache=False)
  File "/usr/share/docassemble/local3.10/lib/python3.10/site-packages/docassemble/webapp/update.py", line 270, in check_for_updates
    except packaging.version.InvalidVersion:
NameError: name 'packaging' is not defined. Did you mean: 'package'?

```

**Comment by japrogramer** [comment]
https://github.com/jhpyle/docassemble/blame/master/docassemble_webapp/docassemble/webapp/update.py#L270

this is the culprit it should read

except version.InvalidVersion:

**Comment by japrogramer** [comment]
```
diff --git a/docassemble_webapp/docassemble/webapp/update.py b/docassemble_webapp/docassemble/webapp/update.py
index 8348938e4..19e27fd86 100644
--- a/docassemble_webapp/docassemble/webapp/update.py
+++ b/docassemble_webapp/docassemble/webapp/update.py
@@ -267,7 +267,7 @@ def check_for_updates(start_time=None, invalidate_cache=True, full=True):
         try:
             pack_vers = version.parse(package.packageversion)
             inst_vers = version.parse(installs[package.id].packageversion)
-        except packaging.version.InvalidVersion:
+        except version.InvalidVersion:
             pack_vers = version.parse('1.0.0')
             inst_vers = version.parse('1.0.0')
         if (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is None) or (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is not None and pack_vers > inst_vers):

```

**Comment by japrogramer** [comment]
@jhpyle this one 

```
diff --git a/docassemble_webapp/docassemble/webapp/update.py b/docassemble_webapp/docassemble/webapp/update.py
index 8348938e4..04b5af860 100644
--- a/docassemble_webapp/docassemble/webapp/update.py
+++ b/docassemble_webapp/docassemble/webapp/update.py
@@ -266,8 +266,11 @@ def check_for_updates(start_time=None, invalidate_cache=True, full=True):
         # logmessage("1: " + str(installs[package.id].packageversion) + " 2: " + str(package.packageversion))
         try:
             pack_vers = version.parse(package.packageversion)
-            inst_vers = version.parse(installs[package.id].packageversion)
-        except packaging.version.InvalidVersion:
+            if package.id in installs and installs[package.id].packageversion is not None:
+                inst_vers = version.parse(installs[package.id].packageversion)
+            else:
+                raise version.InvalidVersion("invalid version")
+        except version.InvalidVersion:
             pack_vers = version.parse('1.0.0')
             inst_vers = version.parse('1.0.0')
         if (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is None) or (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is not None and pack_vers > inst_vers):

```

**Comment by japrogramer** [comment]
diff --git a/docassemble_webapp/docassemble/webapp/update.py b/docassemble_webapp/docassemble/webapp/update.py
index 8348938e4..04b5af860 100644
--- a/docassemble_webapp/docassemble/webapp/update.py
+++ b/docassemble_webapp/docassemble/webapp/update.py
@@ -266,8 +266,11 @@ def check_for_updates(start_time=None, invalidate_cache=True, full=True):
         # logmessage("1: " + str(installs[package.id].packageversion) + " 2: " + str(package.packageversion))
         try:
             pack_vers = version.parse(package.packageversion)
-            inst_vers = version.parse(installs[package.id].packageversion)
-        except packaging.version.InvalidVersion:
+            if package.id in installs and installs[package.id].packageversion is not None:
+                inst_vers = version.parse(installs[package.id].packageversion)
+            else:
+                raise version.InvalidVersion("invalid version")
+        except version.InvalidVersion:
             pack_vers = version.parse('1.0.0')
             inst_vers = version.parse('1.0.0')
         if (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is None) or (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is not None and pack_vers > inst_vers):

**Comment by japrogramer** [comment]
@jhpyle this

```
diff --git a/docassemble_webapp/docassemble/webapp/update.py b/docassemble_webapp/docassemble/webapp/update.py
index 8348938e4..04b5af860 100644
--- a/docassemble_webapp/docassemble/webapp/update.py
+++ b/docassemble_webapp/docassemble/webapp/update.py
@@ -266,8 +266,11 @@ def check_for_updates(start_time=None, invalidate_cache=True, full=True):
         # logmessage("1: " + str(installs[package.id].packageversion) + " 2: " + str(package.packageversion))
         try:
             pack_vers = version.parse(package.packageversion)
-            inst_vers = version.parse(installs[package.id].packageversion)
-        except packaging.version.InvalidVersion:
+            if package.id in installs and installs[package.id].packageversion is not None:
+                inst_vers = version.parse(installs[package.id].packageversion)
+            else:
+                raise version.InvalidVersion("invalid version")
+        except version.InvalidVersion:
             pack_vers = version.parse('1.0.0')
             inst_vers = version.parse('1.0.0')
         if (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is None) or (package.packageversion is not None and package.id in installs and installs[package.id].packageversion is not None and pack_vers > inst_vers):

```

**Comment by jhpyle** [comment]
This was fixed in 1.4.50.
